### PR TITLE
log learning rate

### PR DESCRIPTION
**Pull request by lkhphuc**
Simply log all the learning rates for all parameter groups of all schedulers.

### CODE REVIEW

File: torchtitan/train.py [hunk 1]
```diff
@@ -402,6 +402,9 @@ def main(job_config: JobConfig):
                     "memory/num_alloc_retries": device_mem_stats.num_alloc_retries,
                     "memory/num_ooms": device_mem_stats.num_ooms,
                 }
+                for i_sched, lr_sched in enumerate(lr_schedulers):
+                    for i_group, lr in enumerate(lr_sched.get_last_lr()):
+                        metrics |= { f"lr/scheduler_{i_sched}/group_{i_group}": lr}
```
Comment (fegin) [inline torchtitan/train.py]:
```suggestion
                        metrics[f"lr_scheduler/{i_sched}/group_{i_group}"] = lr
```

File: torchtitan/train.py [hunk 2]
```diff
@@ -402,6 +402,9 @@ def main(job_config: JobConfig):
                     "memory/num_alloc_retries": device_mem_stats.num_alloc_retries,
                     "memory/num_ooms": device_mem_stats.num_ooms,
                 }
+                for i_sched, lr_sched in enumerate(lr_schedulers):
+                    for i_group, lr in enumerate(lr_sched.get_last_lr()):
+                        metrics[f"lr_scheduler/{i_sched}/group_{i_group}"] = lr
```
Comment (tianyu-l) [inline torchtitan/train.py]:
```suggestion
                        metrics[f"learning_rate/{i_sched}/group_{i_group}"] = lr
```

**Review by tianyu-l — COMMENTED** [review]
sounds good to me.

Could you please include a screenshot of the results? Either TB or WandB is fine.

**Review by tianyu-l — COMMENTED** [review]
It seems similar things (with bigger change) are being done in https://github.com/pytorch/torchtitan/pull/938

How about we collaborate over there?

**Comment by lkhphuc** [comment]
Screenshot 2025-03-07 at 10 16 03
Here's a screenshot on WandB (before the log's name change).

> It seems similar things (with bigger change) are being done in #938

Currently that PR does not includes changes to the Exp Tracker, only to the logger. So it's orthogonal to this PR I think. 
I'm not sure how best to proceed but but feel free to include the code in that PR directly before merging. .

**Comment by lkhphuc** [comment]
Ah yes sorry I was confused and missed that part. Those are the same thing. I will close this PR in favor of that PR.
